cmd/antrea-controller/controller.go

```diff
@@ -321,6 +328,9 @@ func run(o *Options) error {
 
 	informerFactory.Start(stopCh)
 	crdInformerFactory.Start(stopCh)
+	if features.DefaultFeatureGate.Enabled(features.AdminNetworkPolicy) {
+		policyInformerFactory.Start(stopCh)
+	}
```

Comment:
I agree that it would pollute the log if the policy API types are not installed. But I think that there are two benefits if we keep logic the current way:
1. For this to happen, as of now, the user has to explicitly enable the AdminNP feature, so having errors indicating that the manifests have not been applied might be helpful
2. If we check the API and then start, it means that when user started Antrea and then installed the AdminNP API types, they will have to restart the Antrea controller before it can work. For now, the process is automatic:
```
W0720 20:14:41.831162       1 reflector.go:424] pkg/mod/k8s.io/client-go@v0.26.4/tools/cache/reflector.go:169: failed to list *v1alpha1.AdminNetworkPolicy: the server could not find the requested resource (get adminnetworkpolicies.policy.networking.k8s.io)
E0720 20:14:41.831215       1 reflector.go:140] pkg/mod/k8s.io/client-go@v0.26.4/tools/cache/reflector.go:169: Failed to watch *v1alpha1.AdminNetworkPolicy: failed to list *v1alpha1.AdminNetworkPolicy: the server could not find the requested resource (get adminnetworkpolicies.policy.networking.k8s.io)
I0720 20:19:59.988098       1 adminnetworkpolicy.go:56] "Processing AdminNetworkPolicy ADD event" anp="pass-example"
```
Please let me know if you think it should be tracked as a TODO, or you would still prefer that we ensure the API types are installed before we start informer

Reply:
I'm more concerned when the feature becomes beta, it would be good to have a TODO in case this is missed in the future.

Reply:
TODO added.